test/fluree/db/transact/transact_test.clj

```diff
@@ -410,3 +410,49 @@
             "schema:name"        "Andrew"
             "schema:description" "He's great!"}
            @(fluree/query db2 {"selectOne" {"ex:andrew" ["*"]}})))))
+
+(deftest ^:integration shacl-datatype-coercion
+  (let [conn   @(fluree/connect {:method :memory})
+        ledger-id "sh-datatype"
+        ledger @(fluree/create conn ledger-id {:defaultContext
+                                               {"ex" "http://example.org/",
+                                                "f" "https://ns.flur.ee/ledger#",
+                                                "rdf" "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
+                                                "rdfs" "http://www.w3.org/2000/01/rdf-schema#",
+                                                "schema" "http://schema.org/",
+                                                "sh" "http://www.w3.org/ns/shacl#",
+                                                "xsd" "http://www.w3.org/2001/XMLSchema#"}})
+
+        db0 (fluree/db ledger)
+        db1 @(fluree/stage2 db0 {"@context" "https://ns.flur.ee",
+                                 "ledger"   ledger-id
+                                 "insert"   {"@id"            "ex:NodeShape/Yeti",
+                                             "@type"          "sh:NodeShape",
+                                             "sh:targetClass" {"@id" "ex:Yeti"},
+                                             "sh:property"    [{"@id"         "ex:PropertyShape/age",
+                                                                "sh:path"     {"@id" "schema:age"},
+                                                                "sh:datatype" {"@id" "xsd:integer"}}]}})
+
+
+        db2 @(fluree/stage2 db1 {"@context" "https://ns.flur.ee",
+                                 "ledger"   ledger-id
+                                 "insert"   {"@id"         "ex:freddy",
+                                             "@type"       "ex:Yeti",
+                                             "schema:name" "Freddy",
+                                             "schema:age"  8}})
+
+        _ @(fluree/commit! ledger db2)
+        loaded @(fluree/load conn ledger-id)
+
+        db3 @(fluree/stage2 (fluree/db loaded) {"@context" "https://ns.flur.ee",
+                                                "ledger" ledger-id
+                                                "insert" {"@id" "ex:letti",
+                                                          "@type" "ex:Yeti",
+                                                          "schema:name" "Letti",
+                                                          "schema:age" "alot"}})]
```

Comment:
needs alignment

Reply:
I'll get it as part of my test overhaul PR :+1: